criu/kerndat.c

```diff
@@ -1736,6 +1737,32 @@ static int kerndat_has_timer_cr_ids(void)
 	return 0;
 }
 
+static int kerndat_has_madv_guard(void)
+{
+	void *map;
+
+	map = mmap(NULL, PAGE_SIZE, PROT_READ | PROT_WRITE, MAP_PRIVATE | MAP_ANONYMOUS, 0, 0);
+	if (map == MAP_FAILED) {
+		pr_perror("Can't mmap a page for has_madv_guard feature test");
+		return -1;
+	}
+
+	if (madvise(map, PAGE_SIZE, MADV_GUARD_INSTALL)) {
+		if (errno != EINVAL) {
+			pr_perror("madvise failed (has_madv_guard check)");
+			goto mmap_cleanup;
+		}
+	}
+
+	munmap(map, PAGE_SIZE);
+	kdat.has_madv_guard = true;
```

Comment:
kdat.has_madv_guard is set to true even when maddvise returns EINVAL. Doesn't look right.

Reply:
Good catch. Fixed, thanks!